File: 671 - Second Minimum Node in a Binary Tree.java
import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private Set<Integer> s = new HashSet<>();
    public int findSecondMinimumValue(TreeNode root) {
        s.clear();
        dfs(root);
        if (s.size() <= 1) {
            return -1;
        } else {
            s.remove(Collections.min(s));
            return Collections.min(s);
        }
    }
    public void dfs(TreeNode root) {
        if (root == null) {
            return;
        }
        s.add(root.val);
        dfs(root.left);
        dfs(root.right);
    }
}

File: 671 - Second Minimum Node in a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    s = set()
    def findSecondMinimumValue(self, root: TreeNode) -> int:
        self.s.clear()
        self.dfs(root)
        if len(self.s) <= 1:
            return -1
        else:
            self.s.remove(min(self.s))
            return min(self.s)
    def dfs(self, root):
        if root is None:
            return
        self.s.add(root.val)
        self.dfs(root.left)
        self.dfs(root.right)